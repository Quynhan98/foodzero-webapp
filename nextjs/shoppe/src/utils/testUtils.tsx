import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
