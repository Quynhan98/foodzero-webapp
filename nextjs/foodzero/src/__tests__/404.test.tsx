import React from 'react'

// Pages
import NotFound from '@pages/404'

// Utils
import { render, screen } from '@utils/testUtils'

describe('NotFound render', () => {
  it('Should show match NotFound DOM Snapshot', () => {
    const { container } = render(<NotFound />)

    expect(container).toMatchSnapshot()
  })

  it('Should navigate to home page', () => {
    render(<NotFound />)

    const link = screen.getByRole('link', {
      name: /back to home/i,
    })

    expect(link.getAttribute('href')).toEqual('/')
  })
})
