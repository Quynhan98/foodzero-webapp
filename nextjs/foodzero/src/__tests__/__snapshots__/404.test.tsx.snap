// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render Should show match NotFound DOM Snapshot 1`] = `
<div>
  <div
    class="css-1wdafdy"
  >
    <p
      class="chakra-text css-1v8em9k"
    >
      404 - Page Not Found!
    </p>
    <a
      href="/"
    >
      <p
        class="chakra-text css-1b5u83m"
      >
        Back To Home
      </p>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
